pkg/kubelet/dockershim/network/cni/cni_others.go

```diff
@@ -49,35 +53,55 @@ func getLoNetwork(binDirs []string) *cniNetwork {
 	return loNetwork
 }
 
-func (plugin *cniNetworkPlugin) platformInit() error {
-	var err error
-	plugin.nsenterPath, err = plugin.execer.LookPath("nsenter")
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
 // TODO: Use the addToNetwork function to obtain the IP of the Pod. That will assume idempotent ADD call to the plugin.
 // Also fix the runtime's call to Status function to be done only in the case that the IP is lost, no need to do periodic calls
 func (plugin *cniNetworkPlugin) GetPodNetworkStatus(namespace string, name string, id kubecontainer.ContainerID) (*network.PodNetworkStatus, error) {
+	podIPs := plugin.getPodIPs(id)
+	if podIPs != nil && len(podIPs) > 0 {
+		klog.V(3).Infof("get pod ip %v from plugin", podIPs)
+		return &network.PodNetworkStatus{IP: podIPs[0], IPs: podIPs}, nil
+	}
+	cninetwork := plugin.getDefaultNetwork()
+	cniNet := cninetwork.CNIConfig
+	netConfList := cninetwork.NetworkConfig
 	netnsPath, err := plugin.host.GetNetNS(id.ID)
 	if err != nil {
 		return nil, fmt.Errorf("CNI failed to retrieve network namespace path: %v", err)
 	}
 	if netnsPath == "" {
 		return nil, fmt.Errorf("cannot find the network namespace, skipping pod network status for container %q", id)
 	}
+	rt, err := plugin.buildCNIRuntimeConf(name, namespace, id, netnsPath, nil, nil)
+	if err != nil {
+		klog.Errorf("Error get pod network status when building cni runtime conf: %v", err)
+		return nil, err
+	}
+	res, err := cniNet.GetNetworkListCachedResult(netConfList, rt)
 
-	ips, err := network.GetPodIPs(plugin.execer, plugin.nsenterPath, netnsPath, network.DefaultInterfaceName)
+	pdesc := format.PodDesc(name, namespace, types.UID(id.ID))
+	if res == nil {
+		klog.V(3).Infof("Cached result doesn't exists for %s", pdesc)
+		return nil, nil
+	}
 	if err != nil {
+		klog.Errorf("Error get cached result %s for network %s: %v", pdesc, netConfList.Name, err)
```

Comment:
How can we get here?  What are the expected cases where `res != nil` and `err != nil` ?

Reply:
probably I just wanted to return err if it existed, yes, it worth merging both condition.